source/Octopus.Tentacle.Client/ITentacleClient.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Client.Scripts;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.ScriptServiceV2;
+
+namespace Octopus.Tentacle.Client
+{
+    public interface ITentacleClient : IDisposable
```

Comment:
As mentioned [here](https://octopusdeploy.slack.com/archives/C04JFJXECS1/p1685666412563579), the Tentacle Client interface might need to be tweaked to be useful for reliable deployments. What is our versioning strategy here? Can we easily evolve it?

Reply:
Do we need to version the TentacleClient? It will be "versioned" through the nuget package version. If it evolves / changes and clients want to make use of the new changes then they will need to bump the package version they reference and make any changes to align with the new client. 

Old clients won't stop working as we will maintain backward compatibility in Tentacle (as we currently do).